lib/urldata.h

```diff
@@ -1916,6 +1916,12 @@ struct Curl_easy {
   iconv_t utf8_cd;             /* for translating to UTF8 */
 #endif /* CURL_DOES_CONVERSIONS && HAVE_ICONV */
   unsigned int magic;          /* set to a CURLEASY_MAGIC_NUMBER */
+#ifdef USE_ARES
+  char *dns_servers;
+  char *dns_interface;
+  char *dns_local_ip4;
+  char *dns_local_ip6;
+#endif /* USE_ARES */
```

Comment:
I think these strings should rather be added in the `str[]` array in the `UserDefined` struct. Those strings already get duplicated...

Reply:
I see! So something like the following then...